File: ts/src/actions/FlowExecution.ts
import type {ActionFlowExecutionResponse} from "@code0-tech/tucana/aquila";
import {CodeZeroEvent} from "../events";
import type {Action} from "../action";

export const packetType = "flowExecutionResponse";

export function handle(action: Action, response: ActionFlowExecutionResponse): void {
    action.emit(CodeZeroEvent.flowExecutionResponseReceived, response);
    action.resolveExecutionResponse(response.executionIdentifier, response.result);
}

File: py/hercules/actions/flow_execution.py
"""Flow execution response handler (port of ``src/actions/FlowExecution.ts``)."""
from __future__ import annotations

from hercules.events import CodeZeroEvent

packet_type = "flow_execution_response"


def handle(action, response) -> None:
    action.emit(CodeZeroEvent.flow_execution_response_received, response)
    action.resolve_execution_response(response.execution_identifier, response)
